File: En C/exercice4_4.c
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <stdlib.h>
#include <sys/wait.h>
int main()
{
    int tube[2] ;
    pipe(tube) ;

    pid_t pid_fils = fork() ;
    if (pid_fils == 0)
    {
        char fd[255];
        sprintf(fd, "%d", tube[0]);
        setenv("TUBE", fd, 1); 
        char *arg[] = {"./fils1", NULL};
        execv("./fils1", arg);
    }
    else
    {
        close(tube[0]);
        write(tube[1], "Bonjour", 8);
    }

    return 0;
}

File: en Python/exercice4_4.py
import os

tube_lecture, tube_ecriture = os.pipe()

pid = os.fork()
if pid == 0:
    os.close(tube_ecriture)
    os.environ["TUBE"] = str(tube_lecture)
    os.execlp("python3", "python3", "fils1.py")

else:
    os.close(tube_lecture)
    os.write(tube_ecriture, b"Bonjour")